src/Tools/dotnet-monitor/Auth/ApiKeyAuthenticationHandler.cs

```diff
@@ -63,11 +66,26 @@ protected override Task<AuthenticateResult> HandleAuthenticateAsync()
                 return Task.FromResult(AuthenticateResult.NoResult());
             }
 
-            //The user is passing a base 64-encoded version of the secret
-            //We will be hash this and compare it to the secret in our configuration.
-            byte[] secret = new byte[ApiKeyAuthenticationHandler.ApiKeyNumBytes];
-            Span<byte> span = new Span<byte>(secret);
-            if (!Convert.TryFromBase64String(authHeader.Parameter, span, out int bytesWritten) || bytesWritten < 32)
+            if (authHeader.Parameter.Length > ApiKeyBase64MaxLength)
+            {
+                return Task.FromResult(AuthenticateResult.Fail("Invalid API key format."));
+            }
+
+            // Calculate the max length of the base64 encoded value,
+            // the rules to decode this are really complex and allow white space in the
+            // string which should be ignored, simply calculate the max it can be.
+            int byteLen = (authHeader.Parameter.Length + 1) * 3 / 4;
```

Comment:
Similarly, use System.Buffers.Text.Base64.GetMaxDecodedFromUtf8Length so we don't have to worry about the algorithm.

Reply:
Done